decidim-core/app/controllers/decidim/newsletters_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Decidim
+  # The controller to handle the user's notifications dashboard.
+  class NewslettersController < Decidim::ApplicationController
+    skip_authorization_check
+
+    layout "decidim/mailer"
+    helper Decidim::SanitizeHelper
+    include Decidim::NewslettersHelper
+
+    def show
+      @organization = current_organization
+      @newsletter = collection.find(params[:id])
+      @user = current_user
+      
+      if @newsletter.sent?
+        @body = parse_interpolations(@newsletter.id, @newsletter.body[I18n.locale.to_s], current_user)
```

Comment:
What if `current_user` does not exist?

Reply:
It makes you login and then unsubscribe you. is this correct?

Reply:
But this action is not for unsubscribing, it's just to show the newsletter from the website. What do you think about only replacing the `%{NAME}` part if `current_user.present?`? You could handle this logic inside the `parse_interpolation` method

Reply:
You are right! Sorry.